life_server/Functions/Systems/fn_cleanup.sqf

```diff
@@ -5,62 +5,44 @@
 
     Description:
     Server-side cleanup script on vehicles.
-    Sort of a lame way but whatever. Yep someone should look at it!
 */
-private "_deleted";
-_deleted = false;
+private _saveFuel = LIFE_SETTINGS(getNumber,"save_vehicle_fuel") isEqualTo 1;
+private _minUnitDistance = LIFE_SETTINGS(getNumber,"vehicles_despawn_max_distance");
+
 for "_i" from 0 to 1 step 0 do {
-    private ["_veh","_units","_fuel"];
     uiSleep (60 * 60);
     {
-        _protect = false;
-        _veh = _x;
-        _vehicleClass = getText(configFile >> "CfgVehicles" >> (typeOf _veh) >> "vehicleClass");
-        _fuel = 1;
-
-        if (!isNil {_veh getVariable "NPC"} && {_veh getVariable "NPC"}) then {_protect = true;};
+        private _vehicleClass = getText(configFile >> "CfgVehicles" >> (typeOf _x) >> "vehicleClass");
+        private _protect = _x getVariable ["NPC",false];
 
         if ((_vehicleClass in ["Car","Air","Ship","Armored","Submarine"]) && {!(_protect)}) then {
-            if (LIFE_SETTINGS(getNumber,"save_vehicle_fuel") isEqualTo 1) then {_fuel = (fuel _veh);};
-            _dbInfo = _veh getVariable ["dbInfo",[]];
-            _units = {(_x distance _veh < 300)} count playableUnits;
-            if (count crew _x isEqualTo 0) then {
-                switch (true) do {
-                    case ((_x getHitPointDamage "HitEngine") > 0.7 && _units isEqualTo 0) : {deleteVehicle _x; _deleted = true;};
-                    case ((_x getHitPointDamage "HitLFWheel") > 0.98 && _units isEqualTo 0) : {deleteVehicle _x; _deleted = true;};
-                    case ((_x getHitPointDamage "HitLF2Wheel") > 0.98 && _units isEqualTo 0) : {deleteVehicle _x; _deleted = true;};
-                    case ((_x getHitPointDamage "HitRFWheel") > 0.98 && _units isEqualTo 0) : {deleteVehicle _x; _deleted = true;};
-                    case ((_x getHitPointDamage "HitRF2Wheel") > 0.98 && _units isEqualTo 0) : {deleteVehicle _x; _deleted = true;};
-                    case (_units isEqualTo 0): {deleteVehicle _x; _deleted = true;};
```

Comment:
Correct me if i'm wrong but your code looks like it would remove alive vehicles, is this intended? It seems the old code specifically checks whether a car is damaged beyond a threshold before allowing it to be deleted

Reply:
Yes it removes alive vehicles if no one is near. Old code did that too: 
`case (_units isEqualTo 0): {deleteVehicle _x; _deleted = true;};` <- deleted cars if no one was near. regardless of damage state